### PR TITLE
dont use hard-coded value for manager namespace

### LINKED ISSUE #1208: Identity controller should not assume manager namespace
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
- Identity controller that creates an AzureIdentityBinding in the same namespace as the `capz-controller-manager` assumes a namespace of `capz-system` 

**What did you expect to happen:**
- users can deploy providers in different namespaces using `clusterctl init` with the `--target-namespace` flag and the identity controller should use that namespace

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2d58abba334517ad59a3f2c7f6d8dd476bfb0f49/api/v1alpha3/types.go#L25  

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by nader-ziada**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
- don't use a hard-coded value for the manager namespace when creating an identity binding
- get namespace form Downward API

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1208

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Remove hardcoded namespace value
```

### CODE REVIEW

**Comment by nprokopic** [comment]
Looks good. Thank you for taking care of this!

File: util/system/namespace.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	NamespaceEnvVarName = "POD_NAMESPACE"
+	// DefaultNamespace is the default value from manifest
+	DefaultNamespace = "capz-system"
+)
+
+// GetManagerNamespace return the namespace where the controller is running
+func GetManagerNamespace() string {
+	managerNamespace := os.Getenv(NamespaceEnvVarName)
+	if managerNamespace == "" {
+		managerNamespace = "capz-system"
```
Comment (CecileRobertMichon) [inline util/system/namespace.go:32]:
```suggestion
		managerNamespace = DefaultNamespace
```

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Comment by nader-ziada** [comment]
@CecileRobertMichon should I back-port this to the release-0.4 branch?

**Comment by CecileRobertMichon** [comment]
Yes please. We have a few other fixes that should be backported so we can aim to cut a patch in the next few days.

**Pull request merged**